sys/mobileapps.js

```diff
@@ -0,0 +1,281 @@
+                `mobileapps.${part}`, rp.title];
+        }
+
+        return hyper.get({
+            uri: new URI(fetchPath),
+        })
+        .catch({ status: 404 }, fetchAndReturnPart);
+    }
+
+    _purgeURIs(hyper, req, revision, purgeLatest) {
+        const rp = req.params;
+        const prefix = `//${rp.domain}/api/rest_v1/page/mobile-sections`;
+        const title = encodeURIComponent(rp.title);
+        let purgeEvents = [
+            {
+                meta: {
+                    uri: `${prefix}/${title}/${revision}`
+                }
+            },
+            {
+                meta: {
+                    uri: `${prefix}-lead/${title}/${revision}`
+                }
+            },
+            {
+                meta: {
+                    uri: `${prefix}-remaining/${title}/${revision}`
+                }
+            }
+        ];
+        if (purgeLatest) {
+            purgeEvents = purgeEvents.concat([
+                {
+                    meta: {
+                        uri: `${prefix}/${title}`
+                    }
+                },
+                {
+                    meta: {
+                        uri: `${prefix}-lead/${title}`
+                    }
+                },
+                {
+                    meta: {
+                        uri: `${prefix}-remaining/${title}`
+                    }
+                },
+
+            ]);
+        }
+
+        return hyper.post({
+            uri: new URI([rp.domain, 'sys', 'events', '']),
+            body: purgeEvents
+        })
+        .catch({ status: 404 }, () => {
+        });
+    }
+
+    _fetchFromMCSAndStore(hyper, req) {
+        const rp = req.params;
+        let serviceURI = `${this._options.host}/${rp.domain}/v1/page/mobile-sections`;
+        serviceURI += `/${encodeURIComponent(rp.title)}`;
+        if (rp.revision) {
+            serviceURI += `/${rp.revision}`;
+        }
+
+        const requests = {
+            newContent: hyper.get({
+                uri: new URI(serviceURI)
+            })
+        };
+
+        if (rp.revision) {
+            // This might be a request to the old revision and we don't want
+            // to issue a purge for the latest revision nor store it in the
+            // key_rev bucket, so check if it's indeed the latest
+            requests.latestRev = hyper.get({
+                // TODO: replace this with the key_rev_value when removing old buckets
+                uri: new URI([rp.domain, 'sys', 'key_value', 'mobileapps.lead', rp.title])
+            })
+            .then((res) => res.body.revision)
+            .catch({ status: 404 }, () => {
+                // We have no revisions for this title, so it's certainly latest.
+                return -1;
+            });
+        } else {
+            requests.latestRev = P.resolve(-1);
+        }
+
+        const shouldStoreNewRev = (latestRev) =>
+            mwUtils.isNoCacheRequest(req) || !rp.revision || rp.revision >= latestRev;
```

Comment:
Removed the check for `no-cache` - it doesn't really make sense to serenader older revisions since we don't store them because of the `latest_hash` policy.